### PR TITLE
Fix/kal selection

### PR DESCRIPTION
**Pull request by MichaelGrosio**

### CODE REVIEW

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-list/kal-list.component.spec.ts [hunk 1]
```diff
@@ -172,18 +173,16 @@ describe('TestListItemComponent', () => {
 
   it('should select an item (single mode)', () => {
     spyOn(listInstances.selectionChange, 'emit');
+    const listCheckbox = fixture.debugElement.queryAll(By.directive(KalCheckboxComponent));
```
Comment (jsp-up) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-list/kal-list.component.spec.ts]: checkboxList

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-list/kal-list.component.ts [hunk 2]
```diff
@@ -8,7 +8,7 @@ import {
   HostListener,
   Input,
   OnChanges,
-  OnDestroy,
+  OnDestroy, OnInit,
```
Comment (jsp-up) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-list/kal-list.component.ts]: imports

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-list/kal-list.component.ts [hunk 3]
```diff
@@ -45,10 +46,12 @@ export interface KalVirtualScrollConfig {
   encapsulation: ViewEncapsulation.None,
   changeDetection: ChangeDetectionStrategy.OnPush
 })
-export class KalListComponent<T> implements CollectionViewer, AfterViewInit, OnChanges, OnDestroy {
+export class KalListComponent<T> implements CollectionViewer, OnInit, AfterViewInit, OnChanges, OnDestroy {
 
   highlighted = null;
 
+  isInitialized = false;
```
Comment (jsp-up) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-list/kal-list.component.ts]: private ?

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-list/kal-list.component.ts [hunk 4]
```diff
@@ -161,7 +177,7 @@ export class KalListComponent<T> implements CollectionViewer, AfterViewInit, OnC
   /**
    * Triggered when selection has changed
    */
-  @Output() selectionChange: EventEmitter<KalSelection<T>> = new EventEmitter<KalSelection<T>>();
+  @Output() selectionChange: EventEmitter<KalSelectionModel<T>> = new EventEmitter<KalSelectionModel<T>>();
```
Comment (jsp-up) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-list/kal-list.component.ts]: readonly

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-list/kal-list.component.ts [hunk 5]
```diff
@@ -267,10 +283,10 @@ export class KalListComponent<T> implements CollectionViewer, AfterViewInit, OnC
...
   selectAll() {
     if (this._selectionMode === KalListSelectionMode.Multiple) {
-      this._selection = new KalSelectionModel({multiple: true, all: true, count: this._selection.count});
+      this._selection.selectAll();
       this.cdr.markForCheck();
 
-      this.selectionChange.emit(this._selection.format());
+      this.selectionChange.emit(this._selection);
     } else {
       throw Error('Cannot select multiple rows width single selection mode.');
```
Comment (jsp-up) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-list/kal-list.component.ts]: with

File: projects/kalidea/kaligraphi/src/lib/molecules/kal-list/kal-list.component.ts [hunk 6]
```diff
@@ -397,6 +413,16 @@ export class KalListComponent<T> implements CollectionViewer, AfterViewInit, OnC
...
   }
 
+  ngOnInit(): void {
+    if (!this._selection) {
+      this._selection = new KalSelectionModel<T>({
+        multiple: this.selectionMode === KalListSelectionMode.Multiple
+      });
+      this.countItems();
+    }
+    this.isInitialized = true;
```
Comment (jsp-up) [inline projects/kalidea/kaligraphi/src/lib/molecules/kal-list/kal-list.component.ts:440]: add spaces

File: projects/kalidea/kaligraphi/src/lib/utils/classes/kal-selection.ts [hunk 7]
```diff
@@ -27,6 +27,11 @@ export class KalSelection<T extends { id?: string }> {
    * Is a multiple selection
    */
   multiple ? = false;
+
+  /**
+   * Total of selection
```
Comment (jsp-up) [inline projects/kalidea/kaligraphi/src/lib/utils/classes/kal-selection.ts]: Total selected elements

**Review by jsp-up — CHANGES_REQUESTED** [review]

**Review by jsp-up — DISMISSED** [review]

**Review by jsp-up — APPROVED** [review]

**Review by ydomenjoud — APPROVED** [review]

**Pull request merged**